openadmet_toolkit/cheminf/data_curation.py

```diff
@@ -0,0 +1,240 @@
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from pydantic import BaseModel, Field, field_validator
+from rdkit import Chem
+from rdkit.rdBase import BlockLogs
+from rdkit_funcs import smiles_to_inchikey, standardize_smiles
+from tqdm import tqdm
+
+
+class CSVProcessing(BaseModel):
+    """
+    Class to handle processing data from a csv downloaded
+    """
+
+    csv_path: Path = Field(..., description="Path to the ChEMBL csv")
+
+    def __init__(self, **data):
+        super().__init__(**data)
+
+    @classmethod
+    def read_csv(csv_path, sep=","):
+        return pd.read_csv(csv_path, sep)
+
+    @classmethod
+    def standardize_smiles_and_convert(self):
+        with BlockLogs():
+            self.chemblData["CANONICAL_SMILES"] = self.chemblData[
+                "Smiles"
+            ].progress_apply(lambda x: standardize_smiles(x))
+        with BlockLogs():
+            self.chemblData["INCHIKEY"] = self.chemblData[
+                "CANONICAL_SMILES"
+            ].progress_apply(lambda x: smiles_to_inchikey(x))
+        self.chemblData.dropna(subset="INCHIKEY", inplace=True)
+
+
+class ChEMBLProcessing(CSVProcessing):
+    """
+    Class to handle processing data from a csv downloaded
+    from ChEMBL
+
+    """
+
+    def __init__(self, **data):
```

Comment:
Rather than taking in the options in the arguments, I would suggest using the factory pattern.

This means options like `pchembl_thresh` `L` etc etc all become class attributes and you can push the processing to just one method (that calls others) to get the job done. 

e.g 

```python
processor = ChEMBLProcessing(pchembl_thresh=0, L=1, csv=blah.csv)
clean_data = processor.process()
```

Where all the gory details can be hidden in `process` if that makes sense?